prime-router/src/main/kotlin/cli/tests/WatersAuthTests.kt

```diff
@@ -22,277 +24,933 @@ import java.time.OffsetDateTime
 import java.util.UUID
 
 /**
- *
+ *  Test a variety of waters endpoints across all our various authorization techniques
  */
-class WatersAuthTests : CoolTest() {
-    override val name = "watersauth"
-    override val description = "Test FHIR Auth"
+class OktaAuthTests : CoolTest() {
```

Comment:
Huge amount of new `./prime test` command-line testing code.  The tests are divided in two:  `server2serverauth` and `oktaauth`.